glrd/util.py

```diff
@@ -43,13 +77,18 @@ def extract_version_data(tag_name):
     return (int(match.group(1)), int(match.group(2))) if match else (None, None)
 
 def get_current_timestamp():
-    """Return the current timestamp."""
-    return int(datetime.now().timestamp())
+    """Get current timestamp in UTC."""
+    return int(datetime.now(pytz.UTC).timestamp())
 
 def timestamp_to_isotime(timestamp):
-    """Convert timestamp to ISO time."""
-    dt = datetime.utcfromtimestamp(float(timestamp))
-    return dt.strftime("%H:%M:%S")
+    """Convert timestamp to ISO time string."""
+    if not timestamp:
+        return 'N/A'
+    try:
+        dt = datetime.fromtimestamp(timestamp, pytz.UTC)
+        return dt.strftime('%H:%M:%S')
+    except (ValueError, TypeError):
+        return 'N/A'
```

Comment:
If this happens data are broken. Not sure if returning `N/A` is really useful here.

Reply:
alright, it now exits